Puyolib/DebugLog.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023, LossRain
+// SPDX-License-Identifier: GPLv3
+
+#pragma once
+
+#include <string>
+#include <deque>
+#include <iostream>
+#include <functional>
+
+
+namespace ppvs {
+
+using std::function;
+
+enum class DebugMessageType {
+	NONE = 0,
+	ERROR = 1,
```

Comment:
My suspicion for why there is an error here on Windows is because something (probably a transitive dependency of `Windows.h`) is defining `ERROR` before including this header. Unfortunately, if it is `Windows.h`, the path of least pain is to just avoid keywords that are defined as Windows preprocessor macros.